File: src/main/java/jukebot/framework/CommandProperties.java
package jukebot.framework;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.TYPE)
public @interface CommandProperties {
    String[] aliases() default {};

    String description() default "No description available";

    CommandCategory category() default CommandCategory.MISC;

    boolean developerOnly() default false;

    boolean nsfw() default false;

    boolean enabled() default true;
}

File: src/main/kotlin/me/devoxin/jukebot/framework/CommandProperties.kt
package me.devoxin.jukebot.framework

@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.CLASS)
annotation class CommandProperties(
    val aliases: Array<String> = [],
    val description: String = "No description available",
    val category: CommandCategory = CommandCategory.MISC,
    val developerOnly: Boolean = false,
    val nsfw: Boolean = false,
    val enabled: Boolean = true
)